autogen/agentchat/conversable_agent.py

```diff
@@ -1654,11 +1806,35 @@ def generate_init_message(self, **context) -> Union[str, Dict]:
         If not overridden, "message" needs to be provided in the context.
 
         Args:
-            **context: any context information, and "message" parameter needs to be provided.
-                       If message is not given, prompt for it via input()
+            **context: any context information. It has the following preserved fields.
+                "message": a str of message. Needs to be provided. Otherwise, input() will be called to get the initial message.
+                "takeaway_method": a string specify the method to extract the takeaway from the chat.
+                    Supported methods are "last_msg" and "reflection_with_llm".
+                    when set "last_msg", it returns the last message of the dialog as the takeaway.
+                    when set "reflection_with_llm", it returns the takeaway extracted using an llm client.
+                    "llm" requires the llm_config to be set in either the sender or the recipient.
+                "takeaway_prompt": a string of text used to prompt a LLM-based agent (the sender or receiver agent) to reflext
+                    on the conversation and extract the key takeaway when takeaway_method is "reflection_with_llm".
+                    Default is None and the following default prompt will be used when "takeaway_method" is set to "reflection_with_llm":
+                    "Identify and extract the final solution to the originally asked question based on the conversation."
+                "carryover": a string or a list of string to specify the carryover information to be passed to this chat. It can be a string or a list of string.
+                    If provided, we will combine this carryover with the "message" content when generating the initial chat
+                    message in `generate_init_message`. Please override the `generate_init_message` if you want to customize
```

Comment:
Likewise, we don't have to mention overriding this method in the docstring.

Reply:
@ekzhu fyi, there are lot of mentions of overriding now in docstr, including the code execution part. If we decide to merge #1405, should the comments about overriding for code execution be removed?

Reply:
Right. I can take a pass and remove the mentioning of overriding related to code execution.